src/core/include/utils/utilities.h

```diff
@@ -135,29 +135,30 @@ inline uint32_t AdditionWithCarryOut(uint64_t a, uint64_t b, uint64_t& c) {
  * any class or struct (ex.: BigInteger, NativeIntegerT, etc.)
  * Ex: auto bitlen = GetIntegerTypeBitLength<short>(); bitlen == 16
  */
-template <typename T,
-          typename std::enable_if<std::is_integral<T>::value && !std::is_same<T, bool>::value, bool>::type = true>
+template <typename T, typename = std::enable_if_t<std::is_integral_v<T> && !std::is_same_v<T, bool>>>
 constexpr usint GetIntegerTypeBitLength() {
     return sizeof(T) * CHAR_BIT;
 }
 
 // TODO (dsuponit): the name of this function Max64BitValue() is misleading as it returns the largest value
 // that can be converted from double to int64_t and not the max value of int64_t. The function must be renamed!!!
 constexpr int64_t Max64BitValue() {
-    // (2^63-1)-(2^10-1) => 2^63-2^10 - max value that could be rounded to int64_t
-    return static_cast<int64_t>((uint64_t(1) << 63) - (uint64_t(1) << 10));
+    // (2^63 - 2^9 - 1) = 9223372036854775295 is max int64_t value convertable to double and back to int64_t
+    // without overflow, but there is loss of precision, which causes a warning when implicitly casted to double.
+    // (2^63-1)-(2^10-1) = 9223372036854774784 = int64_t(double(9223372036854775295))
```

Comment:
Remove the unnecessary comments from the code. Use the comments feature of github instead.